spec/acceptance/mongodb_exporter_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'spec_helper_acceptance'
+
+describe 'prometheus mongodb_exporter' do
+  it 'mongodb_exporter works idempotently with no errors' do
+    pp = 'include prometheus::mongodb_exporter'
+    apply_manifest(pp, catch_failures: true)
+    apply_manifest(pp, expect_changes: true)
+    shell('sleep 30')
+  end
+
+  describe 'prometheus mongodb_exporter version 0.3.1' do
+    it ' mongodb_exporter installs with version 0.3.1' do
+      pp = "class { 'prometheus::mongodb_exporter': version => '0.3.1' }"
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
+      shell('sleep 30')
+    end
+
+    describe process('mongodb_exporter') do
+      its(:args) { is_expected.to match %r{\ -mongodb.uri} }
+    end
+
+    describe service('mongodb_exporter') do
+      it { is_expected.to be_running }
+      it { is_expected.to be_enabled }
+    end
+
+    describe port(9216) do
+      it { is_expected.to be_listening.with('tcp6') }
+    end
+  end
+
+  describe 'prometheus mongodb_exporter version 0.11.0' do
+    it ' mongodb_exporter installs with version 0.11.0' do
+      pp = "class { 'prometheus::mongodb_exporter': version => '0.11.0', use_kingpin => true }"
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
+      shell('sleep 30')
```

Comment:
Could it be this sleep is not working correctly? With systemd I'd suggest some proper ready check. Ideally using the notify type but the exporter probably doesn't support that. An alternative is https://github.com/theforeman/puppet-qpid/commit/7534cd902129921d5f86722ea9e41df7ff188f52

Reply:
That is an interesting thought. 

In the case of ```puppet-jira``, ```shell('sleep x')``` is used quite often and this certainly has a negative effect on the runtime of the build jobs. 

In this case, however, you should only make sure that the service has been started before checking if the port is open.

Reply:
With systemd, a properly written service will not return until it has properly started. Since they often run in the foreground, there is no forking and initialization in the background. The idea of that alternative is that systemd checks the port is open and only then does `systemctl start $service` return.

Reply:
@ekohl that's clever hack!